### PR TITLE
Add integration tests for plan selection

### PR DESCRIPTION
**Pull request by gaetanmaisse**
Reopen a PR for https://github.com/gravitee-io/gravitee-api-management/pull/4545

## Description

Add integration tests for plan selection :

- Only keyless
- Only  api-key
- Only jwt
- Only oauth2
- keyless + api-key
- keyless + api-key + jwt
- keyless + api-key + jwt + oauth2 

---

📚&nbsp;&nbsp;View the storybook of this branch [here](https://612657caa8e859003a8a6430-lljctguexo.chromatic.com)

### CODE REVIEW

**Comment by gaetanmaisse** [comment]
Do not merge without @guillaumelamirand's approval

**Review by guillaumelamirand — APPROVED** [review]

**Pull request merged**
